projects/pgai/pgai/semantic_catalog/search.py

```diff
@@ -31,23 +32,31 @@ async def search_objects(
         config: Configuration for the embedding model used.
         query: Query vector (embedding) to compare against stored object embeddings.
         limit: Maximum number of results to return (default: 5).
+        exclude_ids: ids of objects to exclude from search results
 
     Returns:
         A list of ObjectDescription objects ordered by similarity to the query vector.
     """
     logger.debug(f"searching semantic catalog {catalog_id}")
+    params = dict(query=query, limit=limit)
+    if exclude_ids:
+        params["exclude_ids"] = exclude_ids
     async with con.cursor(row_factory=dict_row) as cur:
         sql = SQL("""\
             select x.*
             from ai.{table} x
-            order by x.{column} <=> %s::vector({dimensions})
-            limit %s
+            {filter}
+            order by x.{column} <=> %(query)s::vector({dimensions})
+            limit %(limit)s
         """).format(
             table=Identifier(f"semantic_catalog_obj_{catalog_id}"),
-            dimensions=SQL(str(int(config.dimensions))),  # pyright: ignore [reportArgumentType]
+            dimensions=Literal(int(config.dimensions)),
```

Comment:
Is doing the `int` cast necessary given that `config.dimensions` is typed as being an int and pydantic should error if any other type is assigned to it?

```suggestion
            dimensions=Literal(config.dimensions),
```